examples/md-flexible/tests/domainDecomposition/MixedBoundaryConditionTest.cpp

```diff
@@ -86,14 +96,24 @@ TEST_F(MixedBoundaryConditionTest, testSimpleReflection) {
 
   EXPECT_NO_THROW(domainDecomposition.reflectParticlesAtBoundaries(autoPasContainer));
 
-  std::vector<std::array<double, 3>> expectedVelocities = {
-      {0.0,1.0,1.0}, {0.0,1.0,1.0}, {0.0,1.0,0.5}, {0.0,1.0,0.5}, {1.0,0.5,0.2},
-      {0.0,-1.0,-1.0}, {0.0,-1.0,-1.0}, {0.0,-1.0,-0.5}, {0.0,-1.0,-0.5}, {-1.0,-0.5,-0.2}
-  };
+  // derive the expected velocities by reflecting the velocities in the appropriate directions
+  std::vector<std::array<double, 3>> expectedVelocities = particleVelocities;
+  expectedVelocities[0][1] *= -1;
+  // particle 1 should not reflect
+  expectedVelocities[2][1] *= -1; expectedVelocities[2][2] *= -1;
+  expectedVelocities[3][1] *= -1;
+  expectedVelocities[4][1] *= -1; expectedVelocities[4][2] *= -1;
+  expectedVelocities[5][1] *= -1;
+  // particle 6 should not reflect
+  expectedVelocities[7][1] *= -1; expectedVelocities[7][2] *= -1;
+  expectedVelocities[8][1] *= -1;
+  expectedVelocities[9][1] *= -1; expectedVelocities[9][2] *= -1;
```

Comment:
Do I see it correctly that there is no test for dimension 0? Maybe add one just for completness.

Reply:
This was mostly a result of using {periodic, reflective, reflective} boundaries in this test. I could add an additional test which tests reflections in all 3 dimensions, should be fairly trivial to write